spec/min_heap_binary_tree_spec.rb

```diff
@@ -78,4 +78,145 @@
       end
     end
   end
+
+  describe "#delete" do
+    it "properly deletes node and keeps min heap state" do
+      initial_node = Node.new 5
+      min_heap_tree = MinHeapBinaryTree.new(initial_node)
+
+      node_7 = Node.new 7
+      node_3 = Node.new 3
+      node_6 = Node.new 6
+
+      min_heap_tree.insert(node_7)
+      min_heap_tree.insert(node_3)
+      min_heap_tree.insert(node_6)
+
+      min_heap_tree.delete(node_6)
+
+      expect(min_heap_tree.tree).to eq(
+        [
+          nil,
+          node_3,
+          node_7,
+          initial_node
+        ]
+      )
+    end
+
+    it "properly deltes a node and keeps min state from level 3" do
```

Comment:
I believe this description is currently used 2 or 3 times. this `it` statement needs to be unique, for each test, describing precisely what you are expecting to happen in that particular spec.